riscv-page-tables/src/page_table.rs

```diff
@@ -658,13 +698,13 @@ impl<T: FirstStagePagingMode> FirstStagePageTable<T> {
         num_pages: u64,
         get_pte_page: &mut dyn FnMut() -> Option<Page<InternalClean>>,
     ) -> Result<FirstStageMapper<T>> {
-        if page_size.is_huge() {
-            return Err(Error::PageSizeNotSupported(page_size));
-        }
-
         let mut inner = self.inner.lock();
-        for a in addr.iter_from().take(num_pages as usize) {
-            inner.lock_4k_leaf_for_mapping(a, get_pte_page)?;
+        for a in addr
+            .iter_from_with_size(page_size)
```

Comment:
Need to check that `addr` is aligned to `page_size` first.

Reply:
fixed